src/Bottle/reset-providers.js

```diff
@@ -11,18 +11,23 @@ var removeProviderMap = function resetProvider(name) {
 };
 
 /**
- * Resets all providers on a bottle instance.
+ * Resets providers on a bottle instance. If 'names' array is provided, only the named providers will be reset.
  *
+ * @param Array names
  * @return void
  */
-var resetProviders = function resetProviders() {
-    var providers = this.originalProviders;
-    Object.keys(this.originalProviders).forEach(function resetPrvider(provider) {
-        var parts = provider.split(DELIMITER);
+var resetProviders = function resetProviders(names) {
+    var tempProviders = this.originalProviders;
+
+    Object.keys(this.originalProviders).forEach(function resetProvider(originalProviderName) {
+        if (Array.isArray(names) && names.indexOf(originalProviderName) === -1) {
```

Comment:
@thomasobrien99 Above, outside this loop, could you add the line:

```js
var shouldFilter = Array.isArray(names);
```

Then you can check the variable here so it is only evaluated once.